core/src/test/kotlin/fr/sncf/osrd/pathfinding/PathfindingSignalingTest.kt

```diff
@@ -0,0 +1,135 @@
+package fr.sncf.osrd.pathfinding
+
+import fr.sncf.osrd.api.pathfinding.request.PathfindingWaypoint
+import fr.sncf.osrd.api.pathfinding.runPathfinding
+import fr.sncf.osrd.graph.Pathfinding
+import fr.sncf.osrd.railjson.schema.common.graph.EdgeDirection
+import fr.sncf.osrd.reporting.exceptions.ErrorType
+import fr.sncf.osrd.reporting.exceptions.OSRDError
+import fr.sncf.osrd.sim_infra.api.BlockId
+import fr.sncf.osrd.train.TestTrains
+import fr.sncf.osrd.utils.DummyInfra
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+import org.assertj.core.api.AssertionsForClassTypes
+import org.assertj.core.api.Assertions as assertjAssertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import fr.sncf.osrd.sim_infra.api.Block
+import org.junit.jupiter.api.BeforeEach
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class PathfindingSignalingTest {
+    private var infra: DummyInfra? = null
+
+    private fun setSigSystemIds(list: List<BlockId>, signalingSystemName: String) {
+        val id = infra!!.fullInfra().signalingSimulator.sigModuleManager.findSignalingSystem(signalingSystemName)
+        list.forEach {
+            infra!!.blockPool[it.index.toInt()].signalingSystemId = id
+        }
+    }
+    @BeforeEach
+    fun setUp() {
+        /*       c1
+                ^  \
+               /    v
+        a --> b     d --> e
+               \    ^
+                v  /
+                 c2
+         */
+        infra = DummyInfra()
+        infra?.addBlock("a", "b")
```

Comment:
In this context you'd prefer `!!` over `?`

But you can also make `infra` not nullable by initializing it to a new instance directly line 23

Reply:
done